File: Java/Carro.java
import java.sql.SQLOutput;

public class Carro {
    public String marca;
    public String modelo;
    public int ano;
    public int velocidade;
    public Motor motor;

    public Carro(String marca, String modelo, int ano,Motor motor) {
        this.marca = marca;
        this.modelo = modelo;
        this.ano = ano;
        this.velocidade = 0;
        this.motor = motor;
    }

    public void acelerar(int incremento){
        this.velocidade += incremento;
        System.out.println("Acelerando...Velocidade atual: "+this.velocidade +" km/h");
    }

    public void frear(int decremento){
        this.velocidade -= decremento;
        if (this.velocidade < 0) {
            this.velocidade = 0; // A velocidade não pode ser negativa
        }
        System.out.println("Freando...Velocidade atual: "+this.velocidade +" km/h");
    }

    public void mostrarVelocidade(){
        System.out.println("Velocidade atual: " + this.velocidade + " km/h");
    }


    public void info(){
        System.out.println("Marca do carro: "+ this.marca);
        System.out.println("Modelo do carro: "+ this.modelo);
        System.out.println("Ano do carro: "+ this.ano);
        motor.infosMotor();


    }


}

File: Go/Carro.go
package listaRicardo

import "fmt"

type Carro struct {
	Marca      string
	Modelo     string
	Ano        int
	Velocidade int
	Motor      Motor
}

func (c *Carro) Acelerar(incremento int) {
	c.Velocidade += incremento
	fmt.Printf("Acelerando... Velocidade atual: %d km/h\n", c.Velocidade)
}

func (c *Carro) Frear(decremento int) {
	c.Velocidade -= decremento
	if c.Velocidade < 0 {
		c.Velocidade = 0
	}
	fmt.Printf("Freando... Velocidade atual: %d km/h\n", c.Velocidade)
}

func (c Carro) MostrarVelocidade() {
	fmt.Printf("Velocidade atual: %d km/h\n", c.Velocidade)
}

func (c Carro) Info() {
	fmt.Printf("Marca do carro: %s\n", c.Marca)
	fmt.Printf("Modelo do carro: %s\n", c.Modelo)
	fmt.Printf("Ano do carro: %d\n", c.Ano)
	c.Motor.InfosMotor()
}

func main() {
	motor := Motor{Estado: "bom", Potencia: 150.0}
	carro := Carro{Marca: "Ford", Modelo: "Mustang", Ano: 2020, Motor: motor}

	carro.Acelerar(50)
	carro.MostrarVelocidade()
	carro.Frear(20)
	carro.MostrarVelocidade()
	carro.Info()
}
